### PR TITLE
feature/ST-173-animated-card-migration

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

**Comment by pgs-gmartowski** [comment]
@james-simpson-ST These are mostly configuration files from stjs-library repo and other necessary files which can be treated as 'core'.

**Comment by pgs-gmartowski** [comment]
@james-simpson-ST Oh, and BWT, could you configure this repo just like stjs-library ? I mean to have at least one approval, pipelines running after commit etc ?

File: src/STCard.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Card from './models/Card/Card';
```
Comment (james-simpson-ST) [inline src/STCard.ts]: This is throwing some tslint warnings which I think would be best fixing now before they get too large and then setting the errors up to emit errors

File: src/STCard.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  private _animatedCardTargetContainer: HTMLDivElement;
+  private _card: Card;
+  private _validation: Validation;
+
+  constructor(config: any) {
+    const {animatedCardContainer} = config;
+    this._animatedCardTargetContainer = document.getElementById(animatedCardContainer) as HTMLDivElement;
+    this._animatedCardTargetContainer.innerHTML = template;
+    this._validation = new Validation();
+    Card.ifCardWrapperExist() && (this._card = new Card(config));
```
Comment (james-simpson-ST) [inline src/STCard.ts]: ifCardWrapperExist has been moved to utils hasn't it?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# JS Library
+
+[![Build Status](https://travis-ci.org/SecureTrading/js-payments.svg?branch=develop)](https://travis-ci.org/SecureTrading/js-payments) <!-- Coveralls --> [![Coverage Status](https://coveralls.io/repos/github/SecureTrading/js-payments/badge.svg?branch=develop)](https://coveralls.io/github/SecureTrading/js-payments?branch=develop) <!-- Browserstack --> [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=eVhLS2ZRY1hqcnJhQXZYVVVZeGhBZnNZL2U4S0ZuNzFDSVk5WFd6MnQvTT0tLVZXd0diVnBPU1RURXJ4S2VrakpuUlE9PQ==--8df058f3f39c1ace5b4d83d27b8c3b46fc3bd90d)](https://www.browserstack.com/automate/public-build/eVhLS2ZRY1hqcnJhQXZYVVVZeGhBZnNZL2U4S0ZuNzFDSVk5WFd6MnQvTT0tLVZXd0diVnBPU1RURXJ4S2VrakpuUlE9PQ==--8df058f3f39c1ace5b4d83d27b8c3b46fc3bd90d) <!-- Dependabot --> [![Dependabot badge](https://img.shields.io/badge/Dependabot-enabled-brightgreen.svg)](https://app.dependabot.com/) <!-- Snyk --> [![Known Vulnerabilities](https://snyk.io/test/github/SecureTrading/js-payments/develop/badge.svg)](https://snyk.io/test/github/SecureTrading/js-payments/develop)
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply README.md]: yes, I hope I've changed every occurrence.
done.

File: __mocks__/libMock.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+const V = window.V;
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply __mocks__/libMock.js]: true, done

File: package.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    "type": "git",
+    "url": "git+https://github.com/SecureTrading/js-payments.git"
+  },
+  "author": "",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/SecureTrading/js-payments/issues"
+  },
+  "homepage": "https://github.com/SecureTrading/js-payments#readme",
+  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply package.json]: I've checked once again and deleted couple of them, but I think rest of them are needed

File: package.json [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  "browserslist": [
+    "> 0.2% and last 2 versions",
+    "not dead",
+    "IE 10"
+  ],
+  "config": {
+    "host": "localhost",
+    "locale": "en_GB"
+  },
+  "scripts": {
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply package.json]: I think those which stay are needed, in core library we had couple more (eg. snyk, testdocker, automated)

File: package.json [hunk 7]
```diff
@@ -0,0 +1,98 @@
+{
+  "name": "js-payments",
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply package.json]: yes, my bad :/ I think now we should have everything with correct prefix

File: src/STCard.ts (hunk 2 above)
Comment (pgs-gmartowski) [reply src/STCard.ts]: yes, actually I deleted most of the implementation stuff and leave only the configuration + core class to start application

File: src/STCard.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/STCard.ts]: looks like it shouldn't appear anymore - imports were not alphabetized

**Pull request merged**